keps/sig-node/5554-in-place-update-pod-resources-alongside-static-cpu-manager-policy/README.md

```diff
@@ -0,0 +1,1210 @@
+* [ ] (R) Production readiness review completed
+* [ ] (R) Production readiness review approved
+* [ ] "Implementation History" section is up-to-date for milestone
+* [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+* [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+This proposal enhances [KEP-1287 (In-place Resource Resize)](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/1287-in-place-update-pod-resources#summary) and [KEP-3570 (CPU Manager)](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/3570-cpumanager) features to support In-place cpu resource resizing of [Pods with Guaranteed Quality of Service](https://kubernetes.io/docs/concepts/workloads/pods/pod-qos/#guaranteed) (QoS) class if all four following conditions are satisfied:
+
+ * In-place Resource Resize is enabled.
+ * Static CPU management policy is enabled.
+ * Topology manager policies can be adhered to.
+ * CPUs allocated upon creation of Guaranteed QoS pods are maintained.
+
+This proposal is integrated with and uses the [KEP-5328 (Node Declared Features framework)](https://github.com/pravk03/enhancements/tree/9de1ba74b4b71883f0ceabf8580d5d817bda36d4/keps/sig-node/5328-node-declared-features), to overcome feature rollout challenges with Version Skew, so additional to above conditions, the following condition is required to be satisfied:
```

Comment:
1. I don't think we should couple this enhancement with Node Declared Features. I think it is OK to say we will try leverage Node Declared Features (if implemented), but I don't want that to be a required dependency unless it is absolutely necessary.

2. You state that this feature is using Node Declared Features to overcome Version Skew challenges, but you've left the Version Skew section below empty. This seems contradictory.

Reply:
cc @pravk03